vfs/vfscache/cache.go

```diff
@@ -526,23 +598,27 @@ func (c *Cache) clean() {
 	fs.Infof(nil, "vfs cache: cleaned: objects %d (was %d) in use %d, to upload %d, uploading %d, total size %v (was %v)", newItems, oldItems, totalInUse, uploadsQueued, uploadsInProgress, newUsed, oldUsed)
 }
 
-// cleaner calls clean at regular intervals
+// cleaner calls clean at regular intervals and upon being kicked for out-of-space condition
 //
 // doesn't return until context is cancelled
 func (c *Cache) cleaner(ctx context.Context) {
+	c.kick = make(chan struct{}, 1)
```

Comment:
I would initialise both of these in the main cache initialization `New` so we know they are valid immediately as the `go c.cleaner()` could conceivably run after something uses `c.kick` or `c.cond`

Reply:
Makes sense!  Will do.